From 4108e7ab83a1da2ebcdc1c8cb21e02e80af62f6d Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Tue, 28 Jul 2026 11:56:23 +0000
Subject: [PATCH] Fix error condition in ssh_ed25519_generate

The call to generate ed25519 keypair can fail, ensure to return error
instead of writing out a key.
---
 ssh-ed25519.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/ssh-ed25519.c b/ssh-ed25519.c
index 2369c3af0..48001ffa1 100644
--- a/ssh-ed25519.c
+++ b/ssh-ed25519.c
@@ -83,7 +83,8 @@ ssh_ed25519_generate(struct sshkey *k, int bits)
 	if ((k->ed25519_pk = malloc(ED25519_PK_SZ)) == NULL ||
 	    (k->ed25519_sk = malloc(ED25519_SK_SZ)) == NULL)
 		return SSH_ERR_ALLOC_FAIL;
-	crypto_sign_ed25519_keypair(k->ed25519_pk, k->ed25519_sk);
+	if (crypto_sign_ed25519_keypair(k->ed25519_pk, k->ed25519_sk) != 0)
+		return SSH_ERR_CRYPTO_ERROR;
 	return 0;
 }
 
-- 
2.51.0

